### PR TITLE
feat: add ignore details [IDE-176]

### PR DESCRIPTION
**Pull request by teodora-sandu**
### Description

Starts using `code-client-go` after https://github.com/snyk/code-client-go/pull/8 gets merged. Also prepares the data for ignored findings to be consumed by IDEs. 

Test it by:
- uncommenting the replace directive, 
- running `make build-debug`,
- adding a replace directive for `snyk-ls` and `code-client-go` in the CLI in the `cliv2` folder
- running `make build` from the CLI 
- running the IntelliJ plugin in debug mode

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

Screenshot 2024-03-12 at 19 12 30

### CODE REVIEW

File: infrastructure/code/convert.go [hunk 1]
```diff
@@ -391,6 +391,21 @@ func (s *SarifConverter) toIssues(baseDir string) (issues []snyk.Issue, err erro
 				CWEs:                testRule.Properties.Cwe,
 			}
 
+			// this can be an array of multiple suppressions in SARIF
+			// but we only store one ignore for now
```
Comment (bastiandoetsch) [inline infrastructure/code/convert.go:395]: Why?

Comment (teodora-sandu) [reply infrastructure/code/convert.go:395]: There can only be one ignore per finding, as far as I know. I'm not sure what would happen though if an ignore expires and the user wants to "renew" it - do they create another on top of it but only one is "active" or would it just get replaced. Maybe I should ask

Comment (teodora-sandu) [reply infrastructure/code/convert.go:395]: https://snyk.slack.com/archives/C06MZKWG0NB/p1710328751971279?thread_ts=1710150868.607839&cid=C06MZKWG0NB

**Review by bastiandoetsch — COMMENTED** [review]
Looks good so far, but I think the tests need some love :)

File: infrastructure/code/convert.go [hunk 2]
```diff
@@ -391,12 +392,46 @@ func (s *SarifConverter) toIssues(baseDir string) (issues []snyk.Issue, err erro
...
 	return issues, errs
 }
 
+func (s *SarifConverter) getIgnoreDetails(result codeClient.Result) (bool, *snyk.IgnoreDetails) {
+	isIgnored := false
+	var ignoreDetails *snyk.IgnoreDetails
+
+	// this can be an array of multiple suppressions in SARIF
+	// but we only store one ignore for now
+	if len(result.Suppressions) == 1 {
```
Comment (bastiandoetsch) [inline infrastructure/code/convert.go:408]: Hm... for resiliency, shouldn't this be `> 0` and then only take the first element for ignoring, maybe logging that there are more?

Comment (teodora-sandu) [reply infrastructure/code/convert.go:408]: That's a good idea!

**Review by bastiandoetsch — APPROVED** [review]
Nice!

**Pull request merged**